iree/hal/cts/cts_test_template.cc.in

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// clang-format off
+#cmakedefine IREE_CTS_TEST_FILE_PATH "@IREE_CTS_TEST_FILE_PATH@"
+#cmakedefine IREE_CTS_DRIVER_REGISTRATION_HDR "@IREE_CTS_DRIVER_REGISTRATION_HDR@"
+#cmakedefine IREE_CTS_DRIVER_REGISTRATION_FN @IREE_CTS_DRIVER_REGISTRATION_FN@
+#cmakedefine IREE_CTS_TEST_CLASS_NAME @IREE_CTS_TEST_CLASS_NAME@
+#cmakedefine IREE_CTS_DRIVER_NAME "@IREE_CTS_DRIVER_NAME@"
```

Comment:
Since all of these are set I think these could just be regular `#define`? Not sure what the general practice is here. In LLVM, at least, it seems they do as I'm suggesting: https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Config/config.h.cmake (e.g. for bug report url). Although maybe what you have is safer because then if something isn't set the macro will be undefined instead of the empty string

Reply:
I'm not sure... I think I'll just keep this as-is until we have more generated files / CMake defines and have more reason to look closer.